### PR TITLE
Return 409 Conflict on non-unique names

### PR DESCRIPTION
**Pull request by imikushin**
Modify APIs to return 409 Conflict on non-unique names where
appropriate.

### CODE REVIEW

File: pkg/event-manager/subscriptions/http_handlers.go [hunk 1]
```diff
@@ -63,7 +62,7 @@ func (h *Handlers) addSubscription(params subscriptionsapi.AddSubscriptionParams
 	defer sp.Finish()
 
 	if err := params.Body.Validate(strfmt.Default); err != nil {
-		return eventsapi.NewEmitEventBadRequest().WithPayload(&models.Error{
```
Comment (imikushin) [inline pkg/event-manager/subscriptions/http_handlers.go]: Looks like this was here by mistake

File: pkg/event-manager/subscriptions/http_handlers.go [hunk 2]
```diff
@@ -74,8 +73,14 @@ func (h *Handlers) addSubscription(params subscriptionsapi.AddSubscriptionParams
...
 	_, err := h.store.Add(s)
 	if err != nil {
+		if entitystore.IsUniqueViolation(err) {
+			return subscriptionsapi.NewAddSubscriptionConflict().WithPayload(&models.Error{
+				Code:    http.StatusConflict,
+				Message: swag.String("error creating subscription: non-unique name"),
+			})
+		}
 		log.Errorf("error when storing the subscription: %+v", err)
-		return eventsapi.NewEmitEventInternalServerError().WithPayload(&models.Error{
```
Comment (imikushin) [inline pkg/event-manager/subscriptions/http_handlers.go]: Looks like this was here by mistake

File: pkg/event-manager/subscriptions/http_handlers.go (hunk 1 above)
Comment (kars7e) [reply pkg/event-manager/subscriptions/http_handlers.go]: Good catch! It would be great if we could unify our models/error responses... maybe some day :)

**Comment by imikushin** [comment]
This is a follow-up on #263

**Pull request merged**